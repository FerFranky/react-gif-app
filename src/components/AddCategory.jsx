import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onAddCategory }) => {
    const [inputValue, setInputValue] = useState('')
    const handleInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        const newValue = inputValue.trim()
        if (newValue <= 1) return
        onAddCategory(newValue)
        setInputValue('')
    }
    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired
}