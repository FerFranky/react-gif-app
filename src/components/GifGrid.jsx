import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs"
import { GiftItem } from "./GiftItem"
import { useFetchGifs } from "../hooks/useFetchGifs"


export const GifGrid = ({ category }) => {
    const { images, isLoading } = useFetchGifs( category )

    return (
        <>
            <h3>{category}</h3>
            {isLoading ? (<h3>Cargando...</h3>):(
            <div className="card-grid">
                {images.map((image) => (
                    <GiftItem 
                    key={ image?.id } 
                    { ...image }
                    />
                ))}
            </div>
            )}
        </>
    )
}
