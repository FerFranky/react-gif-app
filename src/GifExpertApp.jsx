import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon ball'])
    const onAddCategory = (value) => {
        if (categories.includes(value)) return
        setCategories([value, ...categories])
    }
    return (
        <>
            <h1>GifExpertApp</h1>
            
            <AddCategory onAddCategory={ (value) => onAddCategory(value) } />
            {categories.map(category => ( <GifGrid key={category} category={category} /> ))}
        </>
    )
}