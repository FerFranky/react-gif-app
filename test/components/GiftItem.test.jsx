import { GiftItem } from "../../src/components"
import { render, screen } from "@testing-library/react"

describe('Prueba de componente <GiftItem />', () => {
    const title = 'Test componente de Item'
    const url = 'https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=fbf5a697i79gh6m7hg1pydrkz9rwer96zj5aytfoi4991254&ep=v1_gifs_search&rid=giphy.gif&ct=g'

    test('Debe renderizar el snapshot del Item mostrado', () => {
        const { container } = render(<GiftItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar la imagen con la URL indicada y el ALT indicado', () => { 
        render(<GiftItem title={title} url={url} />)
        const {src, alt} = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
     })

     test('Debe mostrar el titulo de componente', () => { 
        render(<GiftItem title={title} url={url} />)
        expect( screen.getByText( title ) ).toBeTruthy()
      })
})