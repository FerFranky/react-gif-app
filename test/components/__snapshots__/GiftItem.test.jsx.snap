// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente <GiftItem /> Debe renderizar el snapshot del Item mostrado 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test componente de Item"
      src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=fbf5a697i79gh6m7hg1pydrkz9rwer96zj5aytfoi4991254&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Test componente de Item
    </p>
  </div>
</div>
`;
